src/copyright/ui/ajax-copyright-hist.php

```diff
@@ -100,7 +100,15 @@ function Output()
     else if($action=="update" || $action=="delete" || $action=="undo")
     {
       $id = GetParm("id", PARM_STRING);
-      list($upload, $item, $hash, $type) = explode(",", $id);
+      $getEachID = explode(",", trim($id, ','));
```

Comment:
Why do you trim first? `"2,3,,"` is for me a valid representation of `["2","3","",""]`?

Reply:
hi, All the four values are required here, explode will consider empty as array value. so i'm using trim before i explode. So that i will not get any empty value.

Please let me know if there is any better approach here.

Reply:
assert that no element is equal to `""`. Your approach can not handle `"2,,,4"` which gets parsed to `["2","","","4"]`

Reply:
To handle this i can use `array_filter` which will omit the empty values and give me the exact count.

Reply:
might be OK but then extract the technical implementation into a function `verifyThatAllFourValuesArePresent($ids)` (or something similarly descriptive).

Reply:
Sure will do it this way.